clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -2631,6 +2632,23 @@ public void showAppInbox(CTInboxStyleConfig styleConfig) {
 
     }
 
+    /**
+     * Dismisses the App Inbox Activity if already opened
+     */
+    public void dismissAppInbox() {
+        try {
+            Activity appInboxActivity = CoreMetaData.getAppInboxActivity();
+            if(appInboxActivity == null) {
+                throw new IllegalStateException("AppInboxActivity reference not found");
+            }
+            Logger.v("Finishing the App Inbox");
+            appInboxActivity.finish();
+        }catch (Throwable t) {
+            Logger.d("Can't dismiss AppInbox, please ensure to call this method after the usage of " +
```

Comment:
use config.getLogger()

Reply:
check if finish throws other exceptions

Reply:
> check if finish throws other exceptions

It may throw an **IllegalStateException** if the activity is already in the process of finishing or has already finished.
To avoid this exception, I've placed `activity.isFinishing()` method before calling the finish(). 

It returns true if the activity is in the process of finishing otherwise false.